test/test_torch.py

```diff
@@ -6079,53 +6101,45 @@ def test_pow(self, device):
                 continue
 
             m1 = torch.empty(0, dtype=dtype, device=device)
-            if m1.is_floating_point():
+            if m1.is_floating_point() or m1.is_complex():
                 m1 = torch.rand(100, 100, dtype=dtype, device=device) + 0.5
             else:
                 # math.pow will overflow and throw exceptions for large integers
                 range_high = 4 if dtype in (torch.int8, torch.uint8) else 10
                 m1 = torch.randint(1, range_high, (100, 100), dtype=dtype, device=device)
 
-            for num in [-2.8, -2, -1, -0.5, 0, 0.5, 1, 2, 3, 4, 3.3]:
-                if isinstance(num, int) and num < 0 and not m1.is_floating_point():
-                    with self.assertRaisesRegex(RuntimeError,
-                                                r'Integers to negative integer powers are not allowed\.'):
-                        torch.pow(m1[4], num)
-                else:
-                    # base - tensor, exponent - number
-                    # contiguous
-                    res1 = torch.pow(m1[4], num)
-                    res2 = res1.clone().zero_()
-                    for i in range(res2.size(0)):
-                        res2[i] = math.pow(m1[4][i], num)
-                    self.assertEqual(res1, res2)
+            # The complex64 implementation in CPU may show some precision issues
+            # when comparing the resulting float value with python double precision
+            # Also using pythons pow instead of math.pow can lead to precision issues with
+            # float32
+            tol_kwargs = {}
 
-                    # non-contiguous
-                    res1 = torch.pow(m1[:, 4], num)
-                    res2 = res1.clone().zero_()
-                    for i in range(res2.size(0)):
-                        res2[i] = math.pow(m1[i, 4], num)
-                    self.assertEqual(res1, res2)
+            exponents = [-2.8, -2, -1, -0.5, 0, 0.5, 1, 2, 3, 4, 3.3]
+            if m1.is_floating_point() or m1.is_complex():
+                exponents += [-2.5j, -1.0j, 0j, 1.0j, 2.5j, 1.0 + 1.0j, -1.0 - 1.5j]
```

Comment:
complex exponents should work with all dtypes and so we should enable the test for all dtypes.
```
exponents = [-2.8, -2, -1, -0.5, 0, 0.5, 1, 2, 3, 4, 3.3]
complex_exponents = [-2.5j, -1.0j, 0j, 1.0j, 2.5j, 1.0 + 1.0j, -1.0 - 1.5j, 3.3j]
if m1.is_complex():
   self._do_pow_for_exponents(m1, exponents + complex_exponents, pow, 10e-4)
else:
   self._do_pow_for_exponents(m1, exponents, math.pow, {})
   self._do_pow_for_exponents(m1, complex_exponents, 10e-4)
```

Reply:
Hey @emcastillo I don't this comment is resolved. Please don't use 10e-4 atol with non-complex tensors and non-complex exponent pair.

let's add the code snippet I mentioned above :D

Reply:
Ooops you are right!
Sorry about this :) I will address it first thing tomorrow morning!

Reply:
I think it should be fine now :)